core/src/main/java/software/amazon/awssdk/core/client/builder/SdkDefaultClientBuilder.java

```diff
@@ -249,22 +251,26 @@ public void setOverrideConfiguration(ClientOverrideConfiguration overrideConfigu
         overrideConfiguration(overrideConfiguration);
     }
 
-    public final B httpConfiguration(ClientHttpConfiguration httpConfiguration) {
-        this.httpConfiguration = httpConfiguration;
+    public final B httpClient(SdkHttpClient httpClient) {
+        this.httpClient = httpClient;
+        this.httpClientBuilder = null;
```

Comment:
probably no need to set `this.httpClientBuilder = null` since we are validating now. same as line 261

Reply:
Good catch.